File: clients/algoliasearch-client-java-2/algoliasearch-core/src/main/java/com/algolia/model/search/EditType.java
package com.algolia.model.search;

import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;

/** Type of edit. */
@JsonAdapter(EditType.Adapter.class)
public enum EditType {
  REMOVE("remove"),

  REPLACE("replace");

  private final String value;

  EditType(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static EditType fromValue(String value) {
    for (EditType b : EditType.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }

  public static class Adapter extends TypeAdapter<EditType> {

    @Override
    public void write(final JsonWriter jsonWriter, final EditType enumeration) throws IOException {
      jsonWriter.value(enumeration.getValue());
    }

    @Override
    public EditType read(final JsonReader jsonReader) throws IOException {
      String value = jsonReader.nextString();
      return EditType.fromValue(value);
    }
  }
}

File: clients/algoliasearch-client-php/lib/Model/Search/EditType.php
<?php

namespace Algolia\AlgoliaSearch\Model\Search;

/**
 * EditType Class Doc Comment
 *
 * @category Class
 * @description Type of edit.
 *
 * @package Algolia\AlgoliaSearch
 */
class EditType
{
    /**
     * Possible values of this enum
     */
    const REMOVE = 'remove';

    const REPLACE = 'replace';

    /**
     * Gets allowable values of the enum
     *
     * @return string[]
     */
    public static function getAllowableEnumValues()
    {
        return [self::REMOVE, self::REPLACE];
    }
}
